cmd/geth/chaincmd.go

```diff
@@ -166,10 +166,12 @@ This command dumps out the state for a given block (or latest, if none provided)
 // initGenesis will initialise the given JSON format genesis file and writes it as
 // the zero'd block (i.e. genesis) or will fail hard if it can't succeed.
 func initGenesis(ctx *cli.Context) error {
-	// Make sure we have a valid genesis JSON
+	if ctx.Args().Len() != 1 {
```

Comment:
Can we not have more arguments? Later on we call `makeConfigNode` which probably expects more args as well

Reply:
Flags and args are different. `ctx.Args()` contains the non-flag arguments. In `geth init`, we only accept one argument, the genesis file.